sw/device/silicon_creator/lib/manifest_def.c

```diff
@@ -9,14 +9,25 @@
 /*
  * Declarations for the manifest fields populdated by the linker script.
  */
-extern const uint32_t _manifest_code_start[];
-extern const uint32_t _manifest_code_end[];
-extern const uint32_t _manifest_entry_point[];
+extern char _manifest_code_start[];
+extern char _manifest_code_end[];
+extern char _manifest_entry_point[];
 
-static_assert(sizeof(uint32_t) == sizeof(uintptr_t), "Pointer is not 32-bits.");
+static_assert(sizeof(uint32_t) == sizeof(char *), "Pointer is not 32-bits.");
 
-const volatile manifest_t kManifest = {
+/**
+ * Manifest definition.
+ *
+ * Declaration of the manifest that resides in the .manifest section. This
+ * should be defined with known values, such as `code_start`, `code_end`, and
+ * `entry_point`, populated. The remaining fields will be updated in the binary
+ * by the signer tool.
```

Comment:
Nitpicking suggestion:

```suggestion
 * Declaration of the manifest that resides in the .manifest section. The
 * initializer should explicitly specify the initial values of the members whose
 * values are known a compilation time, such as `code_start`, `code_end`, and
 * `entry_point`. The remaining fields will be updated in the binary by the
 * signer tool (overriding the implicitly specified initial value of zero).
```

Reply:
Done, thanks!